core/src/main/java/com/expedia/adaptivealerting/core/evaluator/RmseEvaluator.java

```diff
@@ -13,16 +11,46 @@
 
     private int n;
     private double resSumSquares;
+    private double rmse;
 
     public RmseEvaluator() {
         this.n = 0;
         this.resSumSquares = 0;
+        this.rmse = 0;
     }
 
     @Override
-    public double computeScore(double observed, double predicted) {
-        this.resSumSquares += MathUtil.getSquare(observed - predicted);
+    public void update(double observed, double predicted) {
+        double residual = observed - predicted;
+        this.resSumSquares += residual * residual;
         this.n += 1;
-        return MathUtil.getSquareRoot(resSumSquares / n);
+        setRmse(Math.sqrt(resSumSquares / n));
```

Comment:
This doesn't address the issue I mentioned earlier. During an update there is no need to calculate the RMSE at all so just leave this out. It greatly increases the cost of an individual update (relative to the simple updates to resSumSquares and n).